File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ApplicationModule.java
package mx.dev.franco.automusictagfixer.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * The module that provides the context.
 */
@Module
public class ApplicationModule {

    @Singleton
    @Provides
    Context provideContext(Application application) {
        return application;
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ApplicationModule.kt
package mx.dev.franco.automusictagfixer.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * The module that provides the context.
 */
@Module
class ApplicationModule {

    @Singleton
    @Provides
    fun provideContext(application: Application): Context = run {application}

}